GitHub DS Plugin docs (#292)
grafana/github-datasource: docs/sources/setup/provisioning.md

Reviewer: This page specifically focuses on provisioning with the Prometheus Operator. I think that should be more explicit in the page title and introductory text.

Perhaps it's also best suited as a task topic: https://grafana.com/docs/writers-toolkit/structure/topic-types/task/?

Author: done now as discussed in the call

---

> slug: '/provisioning'

Reviewer: ```suggestion
```

title: 'Provisioning'
menuTitle: Provisioning
description: Provisioning the GitHub data source plugin
aliases:
  - github
keywords:
  - data source
  - github
  - github repository
  - API
labels:
  products:
    - oss
    - enterprise
    - cloud
weight: 104
---

# Provisioning

> You can define and configure the GitHub data source in YAML files as part of Grafana’s provisioning system. For more information about provisioning a data source, and for available configuration options, see [Provision Grafana](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources).

Reviewer: We shouldn't use product possessives like "Grafana's": https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanaproductpossessives

```suggestion
You can define and configure the GitHub data source in YAML files with Grafana provisioning.
For more information about provisioning a data source, and for available configuration options, refer to [Provision Grafana](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources).
```

---
title: 'Provisioning'
menuTitle: Provisioning
description: Provisioning the GitHub data source plugin
keywords:
  - data source
  - github
  - github repository
  - API
labels:
  products:
    - oss
    - enterprise
    - cloud
weight: 104
---

# Provisioning

You can define and configure the GitHub data source in YAML files with Grafana provisioning. For more information about provisioning a data source, and for available configuration options, refer to [Provision Grafana](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources).

**Example**

> Here is an example to provision the data source while using the [prometheus-operator](https://github.com/prometheus-operator/prometheus-operator)

Reviewer: ```suggestion
The following YAML snippet can be used to provision the GitHub data source for Grafana if you are using [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator):
```

> ```yaml
> promop:
>   grafana:
>     additionalDataSources:
>       - name: GitHub Repo Insights
>         type: grafana-github-datasource
>         jsonData:
>           owner: ’<repostiory_owner>’
>           repository: ’<repostiory_name>’
>         secureJsonData:
>           accessToken: ’<personal_access_token>’
> ```
> 
> Replace;
> 
> - _`<repostiory_owner>`_: name of the owner who owns the repository
> - _`<repostiory_name>`_: name of the repository to get the data
> - _`<personal_access_token>`_: your GitHub personal access token

Reviewer: ```suggestion
```yaml
promop:
  grafana:
    additionalDataSources:
      - name: GitHub Repo Insights
        type: grafana-github-datasource
        jsonData:
          owner: ’<REPOSITORY_OWNER>’
          repository: ’<REPOSITORY_NAME>’
        secureJsonData:
          accessToken: ’<PERSONAL_ACCESS_TOKEN>’
```

Replace the following variables with your information:

- _`<REPOSITORY_OWNER>`_: name of the owner who owns the repository
- _`<REPOSITORY_NAME>`_: name of the repository to get the data
- _`<PERSONAL_ACCESS_TOKEN>`_: your GitHub personal access token
```
